examples/client/main.cpp

```diff
@@ -43,20 +45,42 @@ unordered_map<string, shared_ptr<PeerConnection>> peerConnectionMap;
 unordered_map<string, shared_ptr<DataChannel>> dataChannelMap;
 
 string localId;
+bool echoDataChannelMessages = false;
 
 shared_ptr<PeerConnection> createPeerConnection(const Configuration &config,
                                                 weak_ptr<WebSocket> wws, string id);
+
+void confirmOnStdout(bool echoed, string id, string type, int length);
 string randomId(size_t length);
 
 int main(int argc, char **argv) {
+	Cmdline *params;
+	try {
+		params = new Cmdline(argc, argv);
+	} catch (const std::range_error&e) {
+		std::cout<< e.what() << '\n';
+		delete params;
```

Comment:
If an exception is thrown in the constructor, the object won't be allocated so you must not delete it.

Reply:
Fixed.